config/config_params.go

```diff
@@ -244,8 +251,43 @@ func (c *Config) InterfacePrefixes() []string {
 	return strings.Split(c.InterfacePrefix, ",")
 }
 
-func (c *Config) InterfaceExcludes() []string {
-	return strings.Split(c.InterfaceExclude, ",")
+// InterfaceExcludes returns a list of Regexp objects, each representing either an
+// exact interface value match or a class of interface matches. This function will
+// panic if any individual value cannot be parsed / compiled into a Regexp.
+func (config *Config) InterfaceExcludes() []*regexp.Regexp {
+	log.Debugf("Compiling regexp values for InterfaceExclude %s \n", config.InterfaceExclude)
+	strValues := strings.Split(config.InterfaceExclude, ",")
+	regexpValues := []*regexp.Regexp{}
+
+	// Helper function to compile the given string value into regexp format, if invalid
+	// log error instead
+	compileAndAdd := func(s string) {
+		r, err := regexp.Compile(s)
+		if err != nil {
+			log.Errorf(
+				"Failed to parse element in list for %v: %v from source %v. %v",
+				"InterfaceExclude", s, config.InterfaceExclude, err)
+		} else {
+			regexpValues = append(regexpValues, r)
+		}
+	}
+
+	for _, strValue := range strValues {
+		// Ignore empty values
+		if len(strValue) == 0 {
+			continue
+		}
+		// Any value not in regexp format will be converted into equivalent regexp
+		// that tests for exact match
+		if !RegexpIfaceElemRegexp.Match([]byte(strValue)) {
+			convertedValue := fmt.Sprintf("^%s$", regexp.QuoteMeta(strValue))
+			compileAndAdd(convertedValue)
+		} else {
+			parsedValue := strValue[1 : len(strValue)-1]
+			compileAndAdd(parsedValue)
+		}
+	}
+	return regexpValues
```

Comment:
Steve, I'm afraid this still isn't how I was expecting - so I guess I've done a poor job of communicating that.  I've posted a commit - which is a delta on top of your changes - at https://github.com/neiljerram/felix/commit/77a59b0baa9616febed27884ebd08161252ee9bb, which I hope will show more clearly what I've been trying to express.

Could you take a look at that and let me know what you think?

Reply:
Hey @neiljerram I just took a look at your changes. 

I see two main changes: 
- Making the parse do more than just validation (so it returns the slice of Regexp objects)
- This makes InterfaceExcludes() redundant, so it's been removed 

I don't think you did a poor job of communicating this as you had a comment about having the Parse return an array of the values instead of the raw (granted this version now returns a slice of Regexp, not just slice of strings). I remember making a comment that I wasn't 100% sure on how that change would affect upstream locations that use `felix` (e.g. FelixConfig), [specifically this change](https://github.com/neiljerram/felix/commit/77a59b0baa9616febed27884ebd08161252ee9bb#diff-5fbf5a733ea1e9a90ce9341bf558beddR154). I recall you said it doesn't affect them and I see what you mean now. The scope of changing that field from string to []*Regexp is still only internal to `felix`.  

If my understanding is correct, your changes look fine to me, shall I just apply your delta on my branch?

Reply:
Yes, I think so, and then I believe we can call this done!

Reply:
Applied and pushed! Was just waiting for CI to pass before I added a comment here confirming. 

Thanks very much for all the feedback @neiljerram! Much appreciated.